### PR TITLE
feat: add `fig-autocomplete` subcommand

### PR DESCRIPTION
**Pull request by fedeci**
## Proposed changes

Add Fig integration to generate a fig spec when `mongocli fig-autocomplete` is run.
This is far by actually providing the full integration with autocomplete repo, but I prefer to split it in smaller PRs that can be merged faster.

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongocli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongocli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/opsmanager/clusters/unmanage.go [hunk 1]
```diff
@@ -75,7 +75,6 @@ func UnmanageBuilder() *cobra.Command {
 	}
 	cmd := &cobra.Command{
 		Use:     "unmanage <name>",
-		Aliases: []string{"rm"},
```
Comment (fedeci) [inline internal/cli/opsmanager/clusters/unmanage.go:78]: This alias name was also aliased in the "delete" subcommand which was shadowing this one, so removing this alias should not be breaking.

File: internal/cli/root/mongocli/builder.go [hunk 2]
```diff
@@ -103,6 +105,7 @@ func Builder(profile *string, argsWithoutProg []string) *cobra.Command {
 		loginCmd,
 		logoutCmd,
 		whoCmd,
+		figautocomplete.Builder(),
```
[the remark this answers is not in the document]
Comment (fedeci) [reply internal/cli/root/mongocli/builder.go:108]: No problem! Will the atlas command no longer be available under `mongocli` after the update?

[the remark this answers is not in the document]
Comment (gssbzn) [reply internal/cli/root/mongocli/builder.go:108]: It will for a while, we may not add any new subcommands to the `mongocli atlas` version

**Review by gssbzn — APPROVED** [review]
Sorry for keep you waiting we were wrapping a couple of other things and I was waiting some confirmations from product about enabling this for mongocli but given https://github.com/withfig/autocomplete/pull/972 was merged I think that ship has sailed now

**Comment by gssbzn** [comment]
I have verified the CLA so merging this

**Comment by boooczek** [comment]
Hi @fedeci I understand this PR is for mongocli. Can we please revert these changes as per my email some time ago? We'll keep it only for the Atlas CLI.

**Comment by fedeci** [comment]
Sure @boooczek!
According to the emails you prefer not to add fig integration to mongocli because it is a further dependency. What do you think about integrating it only in development builds (in this manner it would not be added to deps) so that you can still generate the specs in CI and automatically push them to autocomplete repo?

**Pull request merged**